http_server/request.pony

```diff
@@ -8,23 +8,47 @@ primitive HTTP11 is _Version
   """
   fun string(): String iso^ => recover iso String(8).>append("HTTP/1.1") end
   fun to_bytes(): Array[U8] val => [as U8: 'H'; 'T'; 'T'; 'P'; '/'; '1'; '.'; '1']
+  fun u64(): U64 => 'HTTP/1.1'
   fun eq(o: Version): Bool => o is this
+  fun lt(o: Version): Bool =>
+    match o
+    | let _: HTTP11 => false
+    | let _: HTTP10 => false
+    | let _: HTTP09 => false
+    end
+
 
 primitive HTTP10 is _Version
   """
   HTTP/1.0
   """
   fun string(): String iso^ => recover iso String(8).>append("HTTP/1.0") end
   fun to_bytes(): Array[U8] val => [as U8: 'H'; 'T'; 'T'; 'P'; '/'; '1'; '.'; '0']
+  fun u64(): U64 => 'HTTP/1.0'
```

Comment:
im curious what is going on here.

Reply:
This is a preparation for checking for the correct HTTP version part of a request by checking for a full u64 instead of comparing strings. This is a SWAR technique i saw in another HTTP parser library. The multi-char literal will create the following hex int: `0x485454502F312E30`.

Reply:
Can you document that?